plotting/hits2highLevelEstimations.py

```diff
@@ -0,0 +1,258 @@
+
+hits_collection = detector_dict["hitsCollection"]
+strategy = assumptions_dict["strategy"]
+hit_size = assumptions_dict["hit_size"]
+multipliers = assumptions_dict["multipliers"]
+
+pixel_size_uv = simplify_dict(assumptions_dict["pixel_size_uv"])
+print(f"Pixel size uv: {pixel_size_uv}")
+
+# Update layer related dictionary to have identical keys
+n_cells = simplify_dict(detector_dict["det_element_cells"])
+print("Number of cells: ",n_cells)
+sensor_size_map = simplify_dict(detector_dict["sensor_size_map"])
+sensors_per_module_map = simplify_dict(detector_dict["sensors_per_module_map"])
+print(f"Sensor size: {sensor_size_map}, sensors per module: {sensors_per_module_map}")
+
+if isinstance(hit_size, dict):
+    hit_size_tmp = simplify_dict(hit_size)
+
+    # Convert to defaultdict to avoid KeyErrors
+    # for extra layers not defined in the assumptions
+    # (e.g. the 0 or the N+1 layers)
+    hit_size = defaultdict(int)
+    hit_size.update(hit_size_tmp)
+
+# Pick the right histogram depending on the defined strategy
+h_name = ""
+match strategy:
+    case "hit_counts":
+        h_name = f"h_avg_hits_x_layer_{hits_collection}"
+    case "occupancy":
+        h_name = f"h_avg_occ_x_layer_{hits_collection}"
+    case _:
+        raise AttributeError(f"Unknown strategy defined to compute the bandwidth ({strategy})")
+
+h_bw = input_file.Get(h_name).Clone()
+h_bw.SetNameTitle(f"{input_file_name}_bw_per_layer",f"{input_file_name}_bw_per_layer")
+h_avg_hit_rate = input_file.Get(h_name).Clone()
+h_max_hit_rate = input_file.Get(h_name).Clone() 
+h_max_hit_rate.Reset()
+h_avg_occ_cell_per_layer = input_file.Get(h_name).Clone()
+h_max_cell_occ = input_file.Get(h_name).Clone() 
+h_max_cell_occ.Reset()
+
+layer_cells = {}
+for ln, cells in detector_dict["det_element_cells"].items():
+    layer_cells[ln] = cells
+
+n_layers = len(layer_cells.keys())
+layer_binning = [n_layers, -0.5, n_layers - 0.5]
+if is_endcap(detector_type):
+    print("Endcap detector detected, adjusting layer binning accordingly")
+    max_l = int(n_layers / 2) + 0.5
+    layer_binning = [n_layers + 1, -max_l, +max_l]
+
+
+hist_n_cells = ROOT.TH1D("hist_n_cells", "Number of Cells per Layer;Layer;Number of Cells", *layer_binning) # This is the number of sensors in case of semiconductor detectors
+hist_sensor_size = ROOT.TH1D("hist_sensor_size", "Sensor size per Layer;Layer;Sensor size [mm^{2}]", *layer_binning)
+hist_sensors_per_module = ROOT.TH1D("hist_sensors_per_module", "Sensors per module;Layer;Sensors per module", *layer_binning)
+hist_pixel_size_u = ROOT.TH1D("hist_pixel_size_u", "Pixel Size U per Layer;Layer;Pixel Size U [mm]", *layer_binning)
+hist_pixel_size_v = ROOT.TH1D("hist_pixel_size_v", "Pixel Size V per Layer;Layer;Pixel Size V [mm]", *layer_binning)
+
+# Fill the histograms with data
+for layer, value in n_cells.items():
+    hist_n_cells.SetBinContent(int(layer+1+(len(n_cells)/2 if is_endcap(detector_type) else 0)), value)
```

Comment:
to simplify the reading, I guess something like this should achieve the same result?
```suggestion
for layer, value in n_cells.items():
    hist_n_cells.Fill(layer,  value)
```
(same applies for `hist_sensor_size`, `hist_sensors_per_module` and `hist_pixel_size*` )

Reply:
Thanks, I originally wrote this so that one could loop over the bins, but with the way the code is not this indeed is not needed. Applied